base/Dockerfile

```diff
@@ -1,11 +1,20 @@
-FROM alpine:3.8
-LABEL maintainer="Santiago Torres Arias <santiago@nyu.edu>"
+FROM alpine:3.14 as base-image
 
-RUN apk update && apk add python2 python2-dev py2-pip build-base py2-cffi libressl-dev
+RUN apk update && apk add python3 python3-dev py3-pip build-base py3-cffi libressl-dev libffi-dev
+RUN pip install --upgrade pip && pip install wheel
+WORKDIR /root
+# install latest pinned requirements available from in-toto repo
+# by doing this via '--user' we keep all libs/wheels in ~/.local
+# which can be easily copied into our second stage of the image build
+RUN wget https://github.com/in-toto/in-toto/raw/develop/requirements-pinned.txt
+RUN pip install --user -r requirements-pinned.txt
```

Comment:
why dis?

Reply:
Hey @trishankatdatadog, 

We could skip getting and using the pinned requirements in those two `RUN` steps you've highlighted, and just let pip attempt to auto-resolve these dependencies on its own when we `RUN pip install --user in-toto` but it seemed prudent to leverage those pinned python resources that are already being generated within the in-toto project upstream.

It's purely a way to ensure there's more visibility/control over exactly what dependencies are being installed into the container build and ensuring we're remaining consistent. Otherwise we might pull a newer dependency down from pip that hasn't been through the CI process successfully yet (what generates that pinned-requirements.txt file) and perhaps there's a bug which causes some kind of breakage that has now snuck into the docker build because of it pulling in a not-yet-vetted version of dependency.